backend/capellacollab/projects/toolmodels/diagrams/routes.py

```diff
@@ -67,14 +67,15 @@ async def get_diagram_metadata(
     )
 
 
-@router.get("/{diagram_uuid}", response_class=fastapi.responses.Response)
+@router.get("/{diagram_file_name}", response_class=fastapi.responses.Response)
 async def get_diagram(
-    diagram_uuid: str,
+    diagram_file_name: str,
     handler: git_handler.GitHandler = fastapi.Depends(
         git_injectables.get_git_handler
     ),
     logger: logging.LoggerAdapter = fastapi.Depends(log.get_request_logger),
 ):
+    diagram_uuid = diagram_file_name.rsplit(".", maxsplit=1)[0]
```

Comment:
```suggestion
    diagram_uuid = diagram_file_name.split(".")[0]
```

UUIDs can't contain a `.`. Therefore, we can safely assume that everything after the first "." belongs to the file extension.